Add Delay option

daimhin opened the pull request:
Added Delay option to give number of seconds before disabling a server.
Changed Graceful option to use the Delay value if present on the command.

## Description
Added -Delay option to the command that will give the number of seconds before disabling a server.  Also changed the -Graceful to use the value supplied by a Delay.

## Related Issue
https://github.com/devblackops/NetScaler/issues/98

## Motivation and Context
Adding a missing option that is part of the Nitro API.

## How Has This Been Tested?
I ran the updated version of this cmdlet against several of our NetScalers to disable servers.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

EricSCarr committed:
Add Delay option

Added Delay option to give number of seconds before disabling a server.

Changed Graceful option to use the Delay value if present on the command.

iainbrighton commented on NetScaler/Public/Disable-NSLBServer.ps1:
```diff
@@ -64,6 +67,8 @@ function Disable-NSLBServer {
         [parameter(Mandatory,ValueFromPipeline = $true, ValueFromPipelineByPropertyName)]
         [string[]]$Name = (Read-Host -Prompt 'LB server name'),
         
+        [int]$Delay,
```
It looks like the `-Graceful` and `-Delay` parameters should be mutually exclusive, e.g. use one or the other? If this is the case, we should introduce the `Delay` in a separate parameter set.

daimhin replied:
I have updated the code to use parameter sets that makes the Graceful and Delay options mutually exclusive.

iainbrighton commented on NetScaler/Public/Disable-NSLBServer.ps1:
```diff
@@ -87,7 +92,17 @@ function Disable-NSLBServer {
                             $params.Add('delay', $Graceful)
                         } else {
-                            $params.Add('delay', 0)
+                            if ($PSBoundParameters.ContainsKey('Delay')) {
```
You don't need this check as `[Int]` parameters are always 0 unless otherwise specified. You should just be able to safely use `$params.Add('delay', $Delay)`

daimhin replied:
I simplified this section now that the Graceful and Delay are mutually exclusive.

iainbrighton reviewed (changes requested):
Hi @daimhin - thanks for the PR. Looking at the API and your code, I think we should add a parameter set to make sure only `-Graceful 60` or `-Delay 30` can be used, i.e. they cannot be specified together?

EricSCarr committed:
Add Parameter Sets

- Add Parameter Sets to make the Graceful and Delay options mutually exclusive.  Also cleaned up the code where it checks for the values on these parameters.
